edbdeploy/data/terraform/aws-rds-aurora/environments/aws/aws.tf

```diff
@@ -0,0 +1,212 @@
+variable "ssh_user" {}
+variable "ssh_pub_key" {}
+variable "ssh_priv_key" {}
+variable "custom_security_group_id" {}
+variable "cluster_name" {}
+variable "created_by" {}
+variable "ansible_inventory_yaml_filename" {}
+variable "add_hosts_filename" {}
+variable "barman" {}
+variable "pooler_type" {}
+variable "pooler_local" {}
+variable "hammerdb" {}
+variable "public_cidrblock" {}
+variable "project_tag" {}
+variable "rds_security_group_id" {}
+variable "postgres_server" {}
+variable "pg_version" {}
+
+locals {
+  lnx_ebs_device_names = [
+    "/dev/sdf",
+    "/dev/sdg",
+    "/dev/sdh",
+    "/dev/sdi",
+    "/dev/sdj"
+  ]
+}
+
+locals {
+  lnx_nvme_device_names = [
+    "/dev/nvme1n1",
+    "/dev/nvme2n1",
+    "/dev/nvme3n1",
+    "/dev/nvme4n1",
+    "/dev/nvme5n1",
+  ]
+}
+
+locals {
+  barman_mount_points = [
+    "/var/lib/barman"
+  ]
+}
+
+data "aws_subnet_ids" "selected" {
+  vpc_id = var.vpc_id
+}
+
+resource "aws_key_pair" "key_pair" {
+  key_name   = var.cluster_name
+  public_key = file(var.ssh_pub_key)
+}
+
+resource "aws_instance" "hammerdb_server" {
+  count = var.hammerdb_server["count"]
+
+  ami = var.aws_ami_id
+
+  instance_type          = var.hammerdb_server["instance_type"]
+  key_name               = aws_key_pair.key_pair.id
+  subnet_id              = element(tolist(data.aws_subnet_ids.selected.ids), count.index)
+  vpc_security_group_ids = [var.custom_security_group_id]
+
+  root_block_device {
+    delete_on_termination = "true"
+    volume_size           = var.hammerdb_server["volume"]["size"]
+    volume_type           = var.hammerdb_server["volume"]["type"]
+    iops                  = var.hammerdb_server["volume"]["type"] == "io2" ?  var.hammerdb_server["volume"]["iops"] : var.hammerdb_server["volume"]["type"] == "io1" ? var.hammerdb_server["volume"]["iops"] : null
+  }
+
+  tags = {
+    Name       = format("%s-%s%s", var.cluster_name, "hammerdbserver", count.index + 1)
+    Created_By = var.created_by
+  }
+
+  connection {
+    private_key = file(var.ssh_pub_key)
+  }
+}
+
+resource "aws_instance" "pem_server" {
+  count = var.pem_server["count"]
+
+  ami = var.aws_ami_id
+
+  instance_type          = var.pem_server["instance_type"]
+  key_name               = aws_key_pair.key_pair.id
+  subnet_id              = element(tolist(data.aws_subnet_ids.selected.ids), count.index)
+  vpc_security_group_ids = [var.custom_security_group_id]
+
+  root_block_device {
+    delete_on_termination = "true"
+    volume_size           = var.pem_server["volume"]["size"]
+    volume_type           = var.pem_server["volume"]["type"]
+    iops                  = var.pem_server["volume"]["type"] == "io2" ? var.pem_server["volume"]["iops"] : var.pem_server["volume"]["type"] == "io1" ? var.pem_server["volume"]["iops"] : null
+  }
+
+  tags = {
+    Name       = format("%s-%s%s", var.cluster_name, "pemserver", count.index + 1)
+    Created_By = var.created_by
+  }
+
+  connection {
+    private_key = file(var.ssh_pub_key)
+  }
+}
```

Comment:
Do we need PEM server?

Reply:
No it doesn't - it has a remote mode where an agent can connect to a Postgres server on a different host. Of course, that limits functionality; e.g. you can't stop/start services, and you can't monitor the OS.

Reply:
Good to know, thank you @dpage-edb